openjpa-jdbc/src/main/java/org/apache/openjpa/jdbc/schema/ForeignKey.java

```diff
@@ -183,45 +182,45 @@ public boolean isLogical() {
      */
     public boolean isPrimaryKeyAutoAssigned() {
         if (_autoAssign != null)
-            return _autoAssign;
+            return _autoAssign.booleanValue();
         return isPrimaryKeyAutoAssigned(new ArrayList(3));
     }
 
     /**
-     * Helper to calculate whether this foreign key depends on auto-assigned
+     * Helper to calculate whether this foreign key depends on auto-assigned 
      * columns.  Recurses appropriately if the primary key columns this key
      * joins to are themselves members of a foreign key that is dependent on
-     * auto-assigned columns.  Caches calculated auto-assign value as a side
+     * auto-assigned columns.  Caches calculated auto-assign value as a side 
      * effect.
      *
      * @param seen track seen foreign keys to prevent infinite recursion in
      * the case of foreign key cycles
      */
     private boolean isPrimaryKeyAutoAssigned(List seen) {
-        if (_autoAssign != null)
-            return _autoAssign;
+        if (_autoAssign != null) 
+            return _autoAssign.booleanValue();
 
         Column[] cols = getPrimaryKeyColumns();
         if (cols.length == 0) {
             _autoAssign = Boolean.FALSE;
             return false;
         }
 
-        for (Column column : cols) {
```

Comment:
Could you please clarify why `for-in` loops were replaced with `plain-for` loops in this PR?

Reply:
Yes see my general comment, will update